packages/shortest/src/cli/commands/init.ts

```diff
@@ -36,88 +38,135 @@ initCommand
     new Option("--log-level <level>", "Set logging level").choices(LOG_LEVELS),
   )
   .action(async function () {
-    await executeCommand(this.name(), this.optsWithGlobals(), async () =>
-      executeInitCommand(),
-    );
+    await executeCommand(this.name(), this.optsWithGlobals(), async () => {
+      await executeInitCommand();
+    });
   });
 
 export const executeInitCommand = async () => {
-  console.log(pc.blue("Setting up Shortest..."));
-
-  try {
-    const packageJson = await getPackageJson();
-    if (
-      packageJson?.dependencies?.["@antiwork/shortest"] ||
-      packageJson?.devDependencies?.["@antiwork/shortest"]
-    ) {
-      console.log(pc.green("✔ Package already installed"));
-      return;
-    }
-    console.log("Installing @antiwork/shortest...");
-    const installCmd = await getInstallCmd();
-    execSync(installCmd, { stdio: "inherit" });
-    console.log(pc.green("✔ Dependencies installed"));
-
-    const configPath = join(process.cwd(), CONFIG_FILENAME);
-    const exampleConfigPath = join(
-      fileURLToPath(new URL("../../src", import.meta.url)),
-      `${CONFIG_FILENAME}.example`,
-    );
-
-    const exampleConfig = await readFile(exampleConfigPath, "utf8");
-    await writeFile(configPath, exampleConfig, "utf8");
-    console.log(pc.green(`✔ ${CONFIG_FILENAME} created`));
-
-    const envResult = await addToEnv(process.cwd(), {
-      ANTHROPIC_API_KEY: {
-        value: "your_value_here",
-        comment: "Shortest variables",
+  const tasks = new Listr(
+    [
+      {
+        title: "Checking for existing installation",
+        task: async (ctx, task): Promise<void> => {
+          const packageJson = await getPackageJson();
+          ctx.alreadyInstalled = !!(
+            packageJson?.dependencies?.["@antiwork/shortest"] ||
+            packageJson?.devDependencies?.["@antiwork/shortest"]
+          );
```

Comment:
Preserving existing functionality to exit early if the package is detected. Ref https://github.com/antiwork/shortest/pull/297#discussion_r1935142270